cirq-core/cirq/transformers/eject_phased_paulis.py

```diff
@@ -0,0 +1,342 @@
+# Copyright 2022 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Transformer pass that pushes 180° rotations around axes in the XY plane later in the circuit."""
+
+from typing import Optional, cast, TYPE_CHECKING, Iterable, Tuple, Dict
+import sympy
+import numpy as np
+
+from cirq import circuits, ops, value, protocols
+from cirq.transformers import transformer_api, transformer_primitives
+from cirq.transformers.analytical_decompositions import single_qubit_decompositions
+
+if TYPE_CHECKING:
+    import cirq
+
+
+@transformer_api.transformer
+def eject_phased_paulis(
+    circuit: 'cirq.AbstractCircuit',
+    *,
+    context: Optional['cirq.TransformerContext'] = None,
+    atol: float = 1e-8,
```

Comment:
This seems a little tight, floats are usually only good up to 1e-6 or so.

Reply:
1e-8 tolerance is standard across the codebase, try running `grep -RI '1e-8' --include \*.py`.